app/models/purchase.rb

```diff
@@ -156,7 +156,8 @@ class Purchase < Ekylibre::Record::Base
   validate do
     if invoiced_at
       errors.add(:invoiced_at, :before, restriction: Time.zone.now.l) if invoiced_at > Time.zone.now
-      errors.add(:invoiced_at, :not_opened_financial_year) unless opened_financial_year?
+      errors.add(:invoiced_at, :not_opened_financial_year) unless opened_financial_year? || financial_year_in_closure_preparation?
```

Comment:
Can you wrap this in a `if(closure_preparation?)/else` instead of `unless ... closure_preparation?` + `if closure_preparation?` ?